openff/bespokefit/executor/services/qcgenerator/cache.py

```diff
@@ -50,6 +54,94 @@ def _canonicalize_task(task: _T) -> _T:
     return task
 
 
+def _sha256_hash(contents: str) -> str:
+    return hashlib.sha512(contents.encode()).hexdigest()
+
+
+def _retrieve_cached_task_id(
+    task_hash: str, redis_connection: redis.Redis
+) -> Optional[str]:
+    """Retrieve the task ID of a cached QC task if present in the redis cache"""
+
+    task_id = redis_connection.hget("qcgenerator:task-ids", task_hash)
+
+    return None if task_id is None else task_id.decode()
+
+
+def _cache_task_id(
+    task_id: str, task_type: str, task_hash: str, redis_connection: redis.Redis
+):
+
+    redis_connection.hset("qcgenerator:types", task_id, task_type)
+    # Make sure to only set the hash after the type is set in case the connection
+    # goes down before this information is entered and subsequently discarded.
+    redis_connection.hset("qcgenerator:task-ids", task_hash, task_id)
+
+
+def _compute_torsion_drive_task(
+    task: Torsion1DTask, redis_connection: redis.Redis
+) -> str:
+
+    task_hash = _sha256_hash(task.json())
+    task_id = _retrieve_cached_task_id(task_hash, redis_connection)
+
+    if task_id is not None:
+        return task_id
```

Comment:
I think its safe to assume if we are running this method then the full calculation is not in the cache. We can then use the logic in `cached_compute_task` to do this cache check?